sklearn/tests/test_pipeline.py

```diff
@@ -215,8 +215,6 @@ def test_pipeline_init_tuple():
     pipe.fit(X, y=None)
     pipe.score(X)
 
-    X = np.array([[1, 2]])
-    pipe = Pipeline((('transf', Transf()), ('clf', FitParamT())))
```

Comment:
The X was indeed redundant, but the redefinition of the `pipe` is actually to make sure that `set_params` also works when `fit` is not yet called. 

In the end I put my fix (conversion to list) in the `__init__` and not in the `fit` like Alex did, so it shouldn't matter. But now the test is less robust to a change in the Pipeline implementation.